### PR TITLE
Add data volumes to builds and circleci

### PR DESCRIPTION
**Pull request by cbautista1002**
Do not merge yet. Testing builds with data volumes.

### CODE REVIEW

**Comment by jumanjiman** [comment]
i'm creating two new quay repos:

* iseexchange/tacstack_mysqldata
* iseexchange/tacstack_ftpdata

**Comment by cbautista1002** [comment]
Great!

**Comment by jumanjiman** [comment]
ok, i've created the two quay repos. i'll comment in-line in your diff

File: circle.yml [hunk 1]
```diff
@@ -23,12 +23,24 @@ deployment:
     commands:
       - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
       - docker tag tacstack quay.io/iseexchange/tacstack:latest
+      - docker tag mysqldatavol quay.io/iseexchange/tacstack:mysqldatavol
```
Comment (jumanjiman) [inline circle.yml]: this should be `quay.io/iseexchange/tacstack_mysqldata`

File: circle.yml [hunk 2]
```diff
@@ -23,12 +23,24 @@ deployment:
...
+      - docker tag ftpdatavol quay.io/iseexchange/tacstack:ftpdatavol
```
Comment (jumanjiman) [inline circle.yml]: s/b `quay.io/iseexchange/tacstack_ftpdata`

File: circle.yml (hunk 1 above)
Comment (jumanjiman) [inline circle.yml]: correction: `quay.io/iseexchange/tacstack_mysqldata:latest`

File: circle.yml (hunk 2 above)
Comment (jumanjiman) [inline circle.yml]: correction: `quay.io/iseexchange/tacstack_ftpdata:latest`

Comment (cbautista1002) [inline circle.yml]: :latest ?

Comment (cbautista1002) [inline circle.yml]: I got it. Nevermind.

File: circle.yml [hunk 3]
```diff
@@ -23,12 +23,24 @@ deployment:
...
       - docker push quay.io/iseexchange/tacstack:latest
+      - docker push quay.io/iseexchange/tacstack:mysqldatavol
+      - docker push quay.io/iseexchange/tacstack:ftpdatavol
       - docker tag tacstack quay.io/iseexchange/tacstack:${CIRCLE_SHA1:0:7}
+      - docker tag mysqldatavol quay.io/iseexchange/tacstack:mysqldatavol_${CIRCLE_SHA1:0:7}
+      - docker tag ftpdatavol quay.io/iseexchange/tacstack:ftpdatavol_${CIRCLE_SHA1:0:7}
       - docker push quay.io/iseexchange/tacstack:${CIRCLE_SHA1:0:7}
+      - docker push quay.io/iseexchange/tacstack:mysqldatavol${CIRCLE_SHA1:0:7}
```
Comment (jumanjiman) [inline circle.yml]: the `docker push` commands need to be in the following pseudo-code order:

```
- docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
- <all the pushes for app>
- docker logout
- docker login -e . -u $DOCKER_USER -p $MYSQL_PASS quay.io
- <all the pushes for mysqldata>
- docker logout
- docker login -e . -u $DOCKER_USER -p $FTP_PASS quay.io
- <all the pushes for ftpdata>
```

Comment (cbautista1002) [inline circle.yml]: Oh, I see. I guess that explains the earlier failure?

Comment (jumanjiman) [inline circle.yml]: here's why:

we're technically using access tokens to push, not a user.
the access token has push rights; the rest of us have read-only.

from a workflow perspective, this prevents a person from accidentally pushing directly.
all change goes through CI for audit purposes.

in an emergency, we can always change our access perms.

Comment (jumanjiman) [inline circle.yml]: it also means we can change the access tokens regularly

**Comment by cbautista1002** [comment]
@jumanjiman It failed because of permissions with the new repos

File: circle.yml [hunk 4]
```diff
@@ -21,14 +21,42 @@ deployment:
...
       - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
       - docker tag tacstack quay.io/iseexchange/tacstack:latest
-      - docker push quay.io/iseexchange/tacstack:latest
+      - docker push         quay.io/iseexchange/tacstack:latest
       - docker tag tacstack quay.io/iseexchange/tacstack:${CIRCLE_SHA1:0:7}
-      - docker push quay.io/iseexchange/tacstack:${CIRCLE_SHA1:0:7}
+      - docker push         quay.io/iseexchange/tacstack:${CIRCLE_SHA1:0:7}
+      - docker logout
+
+      - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
```
Comment (jumanjiman) [inline circle.yml]: `s/$DOCKER_PASS/$MYSQL_PASS/`

File: circle.yml [hunk 5]
```diff
@@ -21,14 +21,42 @@ deployment:
...
+      - docker tag mysqldatavol quay.io/iseexchange/tacstack_mysqldata:latest
+      - docker push             quay.io/iseexchange/tacstack_mysqldata:latest
+      - docker tag mysqldatavol quay.io/iseexchange/tacstack_mysqldata:${CIRCLE_SHA1:0:7}
+      - docker push             quay.io/iseexchange/tacstack_mysqldata:${CIRCLE_SHA1:0:7}
+      - docker logout
+
+      - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
```
Comment (jumanjiman) [inline circle.yml]: s`/$DOCKER_PASS/$FTP_PASS/`

File: circle.yml [hunk 6]
```diff
@@ -21,14 +21,42 @@ deployment:
...
   pulls:
     branch: /^(?!master).*$/
     commands:
       - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
       - docker tag tacstack quay.io/iseexchange/tacstack:${CIRCLE_USERNAME}_${CIRCLE_BRANCH/\//_}
-      - docker push quay.io/iseexchange/tacstack:${CIRCLE_USERNAME}_${CIRCLE_BRANCH/\//_}
+      - docker push         quay.io/iseexchange/tacstack:${CIRCLE_USERNAME}_${CIRCLE_BRANCH/\//_}
+      - docker logout
+
+      - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
```
Comment (jumanjiman) [inline circle.yml]: `s/$DOCKER_PASS/$MYSQL_PASS/`

File: circle.yml [hunk 7]
```diff
@@ -21,14 +21,42 @@ deployment:
...
+      - docker tag mysqldatavol quay.io/iseexchange/tacstack_mysqldata:${CIRCLE_USERNAME}_${CIRCLE_BRANCH/\//_}
+      - docker push             quay.io/iseexchange/tacstack_mysqldata:${CIRCLE_USERNAME}_${CIRCLE_BRANCH/\//_}
+      - docker logout
+
+      - docker login -e . -u $DOCKER_USER -p $DOCKER_PASS quay.io
```
Comment (jumanjiman) [inline circle.yml]: `s/$DOCKER_PASS/$FTP_PASS/`

**Pull request merged**